frontend/src/features/projects/common/tabs/tabValidation.ts

```diff
@@ -0,0 +1,74 @@
+import { SPPApprovalTabs, IProject, IProjectTask } from '..';
```

Comment:
This validation logic is also used by ERP and SPL so extracted out to this common file.
The motivation for this logic is that we need a way to validate all of the tabs (since they have required fields) before submission, but also inform the user that there are errors on tabs that are currently not being displayed. the ValidationGroup provides a way of splitting validation out on a per-tab basis, so that the component knows what tab an error belongs to.